src/components/Icon/Icon.tsx

```diff
@@ -34,14 +36,16 @@ function getFontAwesomeIcon(icon: IconType): string {
 interface Props {
   /** The type of icon to display */
   icon: IconType
+  /** Function to execute when user clicks on icon */
+  onClick?: (event: React.MouseEvent<any>) => void
```

Comment:
There should be a test to make sure that the onClick gets passed to the FontAwesomeIcon properly and that the `onClick` prop gets fired when the fontawesome icon is clicked.

Reply:
added